respond/collector_test.go

```diff
@@ -34,10 +41,63 @@ func TestParse(t *testing.T) {
 		Raw: compressed,
 	}
 
-	data, err := res.parse()
+	data, err := res.parse([]CustomFieldConfig{})
 
 	assert.NoError(err)
 	assert.NotNil(data)
 
 	assert.Equal("f81a67a5e9c1", data.Nodeinfo.NodeID)
 }
+
+func TestParseCustomFields(t *testing.T) {
+	assert := assert.New(t)
+
+	// read testdata
+	compressed, err := ioutil.ReadFile("testdata/nodeinfo.flated")
+	assert.Nil(err)
+
+	res := &Response{
+		Raw: compressed,
+	}
+
+	customFields := []CustomFieldConfig{
+		{
+			Name: "my_custom_field",
+			Path: "nodeinfo.hostname",
+		},
+	}
+
+	data, err := res.parse(customFields)
+
+	assert.NoError(err)
+	assert.NotNil(data)
+
```

Comment:
```suggestion
assert.Equal("Trillian", data.NodeInfo.Hostname)
```

(glaub der `struct` was `data.NodeInfo.Hostname`)